tests/Features/SearchTest.php

```diff
@@ -63,7 +63,7 @@ public function testInvalidObjectId(): void
 
         $threadIndexMock->shouldReceive('search')->once()->andReturn([
             'hits' => [
-                ['objectID' => '1'],
+                ['objectID' => 'App\Thread@1'],
```

Comment:
Since we are doing `keyBy()` in the ModelsResolver, it is automatically converting `1` to an integer. See [this](https://implode.io/PmnjXT). `App\Thread@1` is still an invalid objectId though.

Reply:
@mozammil Seems a breaking change to me. Let's make sure we keep the 1 as string, just like normal.

Should we cast to string this line? https://github.com/algolia/scout-extended/blob/66b9d2fce2216059c520c9364c0c0b9a9b8bab89/src/Searchable/ModelsResolver.php#L59

Reply:
This wouldn't solve it. The issue with that `ObjectIdEncrypter` is expecting a string. So, the following would solve it: 

```
foreach ($hits->keys() as $id) {
    $modelClass = ObjectIdEncrypter::decryptSearchable((string) $id);
    $modelKey = ObjectIdEncrypter::decryptSearchableKey((string) $id);
```